portal-impl/src/com/liferay/portlet/asset/util/AssetSearcher.java

```diff
@@ -521,7 +521,7 @@ protected void addSearchNotAnyCategories(
 	protected void addSearchNotAnyKeywords(BooleanFilter queryBooleanFilter)
 		throws Exception {
 
-		String[] notAnyKeywords = _assetEntryQuery.getNotAnyKeywords();
+		String[] notAnyKeywords = _escapeSpecialChars(_assetEntryQuery.getNotAnyKeywords());
 
```

Comment:
I think the setEscape(true) is not a function of StringQuery: https://docs.liferay.com/ce/apps/portal-search/2.0.0/javadocs/com/liferay/portal/search/query/StringQuery.html

Reply:
You could do escape here to avoid loop through keywords one more time 
https://github.com/huynguyen-codeengine/liferay-portal/blob/ac636ac0f1b65786d44e1da4ce0d7719c8fa87c8/portal-impl/src/com/liferay/portlet/asset/util/AssetSearcher.java#L536

Reply:
I updated. thank you so much.